PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -2783,6 +2782,48 @@ intptr_t ScintillaEditView::searchInTarget(const wchar_t * text2Find, size_t len
 	return execute(SCI_SEARCHINTARGET, len, reinterpret_cast<LPARAM>(text2FindA));
 }
 
+intptr_t ScintillaEditView::searchInHtml(const wchar_t* text2Find,size_t fromPos,size_t toPos,bool reverse) const
+{
+	// Convert types for efficient runtime
+	std::wstring wstr = text2Find;
+	int size_needed = WideCharToMultiByte(CP_UTF8, 0, wstr.c_str(), -1, nullptr, 0, nullptr, nullptr);
+	std::string utf8Str(size_needed, 0);
+	WideCharToMultiByte(CP_UTF8, 0, wstr.c_str(), -1, &utf8Str[0], size_needed, nullptr, nullptr);
+	if (!utf8Str.c_str())
+	{
+		return -1;
+	}
+	// Do a search for the pattern
+	Sci_TextToFind ttf;
+	ttf.chrg.cpMin = static_cast<Sci_PositionCR>(fromPos);

```

Comment:
Use `Sci_TextToFind`&#x200D;__*Full*__ and `SCI_FINDTEXT`&#x200D;__*FULL*__ instead: <https://groups.google.com/g/scintilla-interest/c/mPLwYdC0-FE>

Although `Sci_PositionCR` is effectively the same size as `Sci_Position` ([since N++ 8.3][0]), the Scintilla definition should be preferred.

---

@chcg, next time there's a Scintilla upgrade, maybe the headers can be edited to include the deprecation warnings currently in the plugin template?, e.g.: https://github.com/npp-plugins/plugintemplate/blob/319c80b1161b332a203b7c6df2cadca6d68a9a88/src/Scintilla.h#L1340

[0]: https://github.com/notepad-plus-plus/notepad-plus-plus/commit/db33083

Reply:
@rdipardo I had some branch for it since some time (https://github.com/chcg/notepad-plus-plus/tree/deprecate_32bit_sci_structs_and_defines) , but not pushed it so far.

Reply:
@chcg , is that something that could be pushed?  @rdipardo suggested we wait on https://github.com/npp-plugins/plugintemplate/pull/14 until the deprecation is in the N++ repo, since it will start automatically updating the plugin template from the N++ repo... And we don't want the template to lose its existing deprecation change